utils/etherspot.js

```diff
@@ -62,37 +73,54 @@ export async function investByAAWallet(investmentAmount, chosenToken) {
   // console.log("\x1b[33m%s\x1b[0m", `EtherspotWallet balances:`, balances);
 }
 
-async function getPortfolioHelper(portfolioName) {
+async function getPortfolioHelper(portfolioName, walletClient, chain) {
   let portfolioHelper;
   if (portfolioName === "AllWeatherPortfolio") {
-    portfolioHelper = new AllWeatherPortfolio();
+    portfolioHelper = new AllWeatherPortfolio(walletClient, chain);
   }
   await portfolioHelper.initialize();
   return portfolioHelper;
 }
 
 export class AllWeatherPortfolio {
-  constructor() {
+  constructor(walletClient, chain) {
     // initializating sdk...
-    const customBundlerUrl = "";
-    this.primeSdk = new PrimeSdk(
-      { privateKey: process.env.NEXT_PUBLIC_WALLET_PRIVATE_KEY },
-      {
-        chainId: Number(process.env.NEXT_PUBLIC_CHAIN_ID),
-        projectKey: "all-weather-dev",
-        bundlerProvider: new EtherspotBundler(
-          Number(process.env.NEXT_PUBLIC_CHAIN_ID),
-          process.env.NEXT_PUBLIC_ETHERSPOT_PROJECT_KEY,
-          customBundlerUrl,
-        ),
+    this.AAWallet = new SmartAccount(walletClient, {
```

Comment:
this is how I pass `walletClient` into Particle's AA wallet. If you get a chance, would you tell me how can I pass the correct wallet object from wagmi/viem into `PrimeSdk`? There's no public interface to get the private key from these 2 libraries

```js
import { useConnect } from "wagmi";
const { data: walletClient } = useWalletClient();
```